server/v2/cometbft/abci.go

```diff
@@ -325,7 +329,7 @@ func (c *Consensus[T]) PrepareProposal(
 			continue
 		}
 
-		decodedTxs[i] = decTx
+		decodedTxs = append(decodedTxs, decTx)
```

Comment:
i dont believe this is correct. in the case of vote extensions we will not be able to decode the txs meaning we will always remove it from the block. This should happen in the handler instead of here. This is how its handled in abci.